### PR TITLE
CSHARP-4652: TagSet.Equals should ignore tags order

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver.Core/TagSet.cs [hunk 1]
```diff
@@ -46,7 +44,7 @@ public TagSet()
         /// <param name="tags">The tags.</param>
         public TagSet(IEnumerable<Tag> tags)
         {
-            _tags = Ensure.IsNotNull(tags, nameof(tags)).ToList();
+            _tags = Ensure.IsNotNull(tags, nameof(tags)).OrderBy(t => t.Name).ToArray();
```
Comment (JamesKovacs) [inline src/MongoDB.Driver.Core/TagSet.cs:47]: Tag names aren't guaranteed to be unique. For example a node could have a `TagSet` of `{ usage: "operational", usage: "reporting" }`. This will not match `{ usage: "reporting", usage: "operational" }`, but it should.

Add `ThenBy(t => t.Value)` as well as a test for this edge case.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Found an edge case that needs addressing.

**Review by JamesKovacs — APPROVED** [review]
LGTM
